lib/ruby_lsp/listener.rb

```diff
@@ -19,11 +19,18 @@ def initialize(message_queue)
       @message_queue = message_queue
     end
 
+    @event_to_listener_map = T.let(Hash.new { |h, k| h[k] = [] }, T::Hash[Symbol, T::Array[T.class_of(Listener)]])
+
+    sig { params(event: Symbol).returns(T.nilable(T::Boolean)) }
```

Comment:
(nit) Worth removing the possibility of a nil here? e.g.

```
    sig { params(event: Symbol).returns(T::Boolean) }
    def registered_for_event?(event)
      Listener.event_to_listener_map[event]&.include?(self.class) || false
    end
```

Reply:
Not sure there's a lot of benefit other than the small simplification in the signature.